src/backends/blob.rs

```diff
@@ -0,0 +1,95 @@
+use crate::Krate;
+use anyhow::{Context, Error};
+use azure_sdk_core::prelude::*;
+use azure_sdk_storage_blob::prelude::*;
+use azure_sdk_storage_core::{client, key_client::KeyClient};
+use bytes::Bytes;
+
+#[derive(Debug)]
+pub struct BLOBBackend {
+    prefix: String,
+    client: KeyClient,
+    container: String,
+}
+
+impl BLOBBackend {
+    pub async fn new(
+        loc: crate::BlobLocation<'_>,
+        account: String,
+        master_key: String,
+    ) -> Result<Self, Error> {
+        let client = client::with_access_key(&account, &master_key);
+        Ok(Self {
+            prefix: loc.prefix.to_owned(),
+            container: loc.container.to_owned(),
+            client,
+        })
+    }
+    #[inline]
+    fn make_key(&self, krate: &Krate) -> String {
+        format!("{}{}", self.prefix, krate.cloud_id())
+    }
+}
+
+#[async_trait::async_trait]
+impl crate::Backend for BLOBBackend {
+    async fn fetch(&self, krate: &Krate) -> Result<Bytes, Error> {
+        let response = self
+            .client
+            .get_blob()
+            .with_container_name(&self.container)
+            .with_blob_name(&self.make_key(krate))
+            .finalize()
+            .await
+            .context("failed to fetch object")?;
+        Ok(response.data.into())
+    }
+    async fn upload(&self, source: Bytes, krate: &Krate) -> Result<usize, Error> {
+        let len = source.len();
+        let digest = md5::compute(&source[..]);
+        self.client
+            .put_block_blob()
+            .with_container_name(&self.container)
+            .with_blob_name(&self.make_key(krate))
+            .with_content_type("text/plain")
```

Comment:
These are compressed tarballs, not text, was there a reason for setting this content type?

Reply:
Sorry, it is just as a placeholder. Should it be `application/x-tar`?

Reply:
Yes, thanks!

Reply:
Ok. Let me add a commit. Thanks for your patient.